src/struct/commands/arguments/TypeResolver.js

```diff
@@ -122,6 +122,58 @@ class TypeResolver {
                 return color;
             },
 
+            [ArgumentTypes.TIME]: (message, phrase) => {
+                if (!phrase) return null;
+
+                const TIME_UNITS = {
+                    years: {
+                        label: '(?:y|years?|year?)',
+                        value: 1000 * 60 * 60 * 24 * 365,
+                    },
+                    weeks: {
+                        label: '(?:w|weeks?|week?)',
+                        value: 1000 * 60 * 60 * 24 * 7,
+                    },
+                    days: {
+                        label: '(?:d|days?|day?)',
+                        value: 1000 * 60 * 60 * 24,
+                    },
+                    hours: {
+                        label: '(?:h|hours?|hrs?)',
+                        value: 1000 * 60 * 60,
+                    },
+                    minutes: {
+                        label: '(?:m|minutes?|mins?)',
+                        value: 1000 * 60,
+                    },
+                    seconds: {
+                        label: '(?:s|seconds?|secs?)',
+                        value: 1000,
+                    },
+                    months: {
+                        label: '(?:mo|months?|month?)',
+                        value: 1000 * 60 * 60 * 24 * 30,
+                    },
+                    milliseconds: {
+                        label: '(?:ms|milliseconds?|msecs?)',
+                        value: 1,
+                    },
+                };
+
+                const regexString = Object.entries(TIME_UNITS).map(([name, { label }]) => `(?:(?<${name}>-?(?:\\d+)?\\.?\\d+) *${label})?`).join('\\s?');
+
+                const match = new RegExp(`^${regexString}$`, 'i').exec(phrase);
+                if (!match) return null;
+
+                let milliseconds = 0;
+                for (const key in match.groups) {
+                    const value = Number(match.groups[key] || 0);
+                    milliseconds += value * TIME_UNITS[key].value;
+                }
```

Comment:
I'm generally against using `for-in` anywhere, though it's safe to do so here. You _could_ rewrite this code using `Object.entries`, will leave it up to you to decide of course but personally I feel this is better:

```suggestion
                for (const [key, amount] of Object.entries(match.groups)) {
                    const value = Number(amount || 0);
                    milliseconds += value * TIME_UNITS[key].value;
                }
```